src/main/java/com/tabnine/statusBar/TabnineStatusBarWidget.java

```diff
@@ -1,31 +1,54 @@
 package com.tabnine.statusBar;
 
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.application.ModalityState;
 import com.intellij.openapi.editor.colors.EditorColorsManager;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.wm.CustomStatusBarWidget;
+import com.intellij.openapi.wm.StatusBar;
+import com.intellij.openapi.wm.WindowManager;
 import com.intellij.openapi.wm.impl.status.EditorBasedWidget;
 import com.intellij.openapi.wm.impl.status.TextPanel;
 import com.intellij.util.Consumer;
 import com.tabnine.binary.BinaryRequestFacade;
 import com.tabnine.binary.requests.config.ConfigRequest;
+import com.tabnine.binary.requests.config.StateResponse;
 import com.tabnine.binary.requests.statusBar.ConfigOpenedFromStatusBarRequest;
+import com.tabnine.general.ServiceLevel;
+import com.tabnine.general.StaticConfig;
+import com.tabnine.lifecycle.BinaryStateChangeNotifier;
+import com.tabnine.lifecycle.BinaryStateService;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 import javax.swing.*;
+import java.awt.*;
 import java.awt.event.MouseAdapter;
 import java.awt.event.MouseEvent;
 import java.util.Objects;
 
 import static com.tabnine.general.StaticConfig.ICON_AND_NAME;
 import static com.tabnine.general.StaticConfig.ICON_AND_NAME_DARK;
 
-public class TabnineStatusBarWidget extends EditorBasedWidget implements CustomStatusBarWidget, com.intellij.openapi.wm.StatusBarWidget.WidgetPresentation {
+public class TabnineStatusBarWidget extends EditorBasedWidget implements CustomStatusBarWidget,
+        com.intellij.openapi.wm.StatusBarWidget.WidgetPresentation {
     private final BinaryRequestFacade binaryRequestFacade;
+    private volatile ServiceLevel serviceLevel;
+    private TextPanel.WithIconAndArrows component;
 
     public TabnineStatusBarWidget(@NotNull Project project, BinaryRequestFacade binaryRequestFacade) {
         super(project);
         this.binaryRequestFacade = binaryRequestFacade;
+        final BinaryStateService binaryStateService = ApplicationManager.getApplication().getComponent(BinaryStateService.class);
```

Comment:
Why not using `ServiceManager.getService()`? It's a service, not a component

Reply:
My bad. I was trying to use ApplicationManager.getService which was only added in 2019.3 so I thought this was the only alternative. Missed ServiceManager.
Changed.